Source/utils/file_util.cpp

```diff
@@ -472,4 +475,64 @@ FILE *OpenFile(const char *path, const char *mode)
 #endif
 }
 
+std::vector<std::string> ListDirectories(const char *path)
+{
+	std::vector<std::string> dirs;
+#ifdef NXDK
+	// Use the NXDK Windows API.
+	WIN32_FIND_DATAA findData;
+	// Construct the search path by appending the directory separator and wildcard.
+	std::string searchPath = std::string(path) + DIRECTORY_SEPARATOR_STR + "*";
+	HANDLE hFind = FindFirstFileA(searchPath.c_str(), &findData);
+	if (hFind == INVALID_HANDLE_VALUE)
+		return dirs;
+	do {
+		std::string folder = findData.cFileName;
+		// Skip the special entries "." and ".."
+		if (folder == "." || folder == "..")
+			continue;
+		if (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+			dirs.push_back(folder);
+	} while (FindNextFileA(hFind, &findData));
+	FindClose(hFind);
+#else
+	// For platforms that support std::filesystem.
+	for (const auto &entry : std::filesystem::directory_iterator(path)) {
+		if (entry.is_directory())
+			dirs.push_back(entry.path().filename().string());
+	}
+#endif
+	return dirs;
+}
+
+std::vector<std::string> ListFiles(const char *path)
+{
+	std::vector<std::string> files;
+#ifdef NXDK
+	// Use the NXDK Windows API.
+	WIN32_FIND_DATAA findData;
+	// Construct the search path by appending the directory separator and wildcard.
+	std::string searchPath = std::string(path) + DIRECTORY_SEPARATOR_STR + "*";
+	HANDLE hFind = FindFirstFileA(searchPath.c_str(), &findData);
+	if (hFind == INVALID_HANDLE_VALUE)
+		return files;
+	do {
+		std::string file = findData.cFileName;
+		// Skip the special entries "." and ".."
+		if (file == "." || file == "..")
+			continue;
+		if (!(findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY))
+			files.push_back(file);
+	} while (FindNextFileA(hFind, &findData));
+	FindClose(hFind);
+#else
+	// For platforms that support std::filesystem.
+	for (const auto &entry : std::filesystem::directory_iterator(path)) {
+		if (entry.is_regular_file())
+			files.push_back(entry.path().filename().string());

```

Comment:
We should support Unicode characters in file paths. Maybe something like this.
```suggestion
		if (!entry.is_directory())
			continue;
		std::u8string filename = entry.path().filename().u8string();
		dirs.emplace_back(filename.begin(), filename.end());
```